File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        my_dict = {}
        for i in range(len(nums)):
            if nums[i] in my_dict:
                my_dict[nums[i]] += 1
            else:
                my_dict[nums[i]] = 1
        print(my_dict)
                
        amount = k
        my_list = []

        while amount > 0:
            my_num = 0
            count = 0
            for key, value in my_dict.items():
                if value > count:
                    my_num = key
                    count = value
            my_list.append(my_num)
            amount -= 1
            del my_dict[my_num]
        return my_list


        

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_set<int> c;
        for (int i = 0; i < nums.size(); i++) {
            if (c.count(nums[i])) {
                return nums[i];
            }
            else c.insert(nums[i]);
        }
    }
};
